mobile/native/src/ln_dlc/channel_status.rs

```diff
@@ -0,0 +1,197 @@
+    /// there was a previous channel nor does it imply that a previous channel was completely
+    /// closed i.e. there might be pending transactions.
+    NotOpen,
+    /// There is a vanilla Lightning channel, without a subchannel.1
+    ///
+    /// This means that it is possible to use the Lightning channel for payments and that the
+    /// channel can be upgraded to an LN-DLC channel.
+    LnOpen,
+    /// There is an LN-DLC channel.
+    ///
+    /// This corresponds to an app which currently has an open position as a result of trading.
+    LnDlcOpen,
+    // We cannot easily model the vanilla Lightning channel being force-closed because LDK erases
+    // a closed channel from the `ChannelManager` as soon as the closing process begins. This is a
+    // problem because it's not safe to delete the app if the channel is not fully closed!
+    //
+    // TODO: Add support for `LnForceClosing` status and ensure that channel status remains in
+    // `LnDlcForceClosing` whilst the LN channel is still closing.
+    // LnForceClosing,
+    /// The LN-DLC channel is in the process of being force-closed.
+    LnDlcForceClosing,
+    /// The LN-DLC channel is in an inconsistent state.
+    Inconsistent,
+}
+
+pub async fn track_channel_status(node: impl Borrow<Node>) {
+    loop {
+        tracing::info!("Tracking channel status");
+
+        let status = channel_status(node.borrow())
+            .await
+            .map_err(|e| {
+                tracing::error!("Could not compute LN-DLC channel status: {e:#}");
+                e
+            })
+            .ok();
+
+        tracing::info!(?status, "Channel status udpate");
+
+        event::publish(&event::EventInternal::ChannelStatusUpdate(status));
+
+        tokio::time::sleep(UPDATE_CHANNEL_STATUS_INTERVAL).await;
+    }
+}
+
+/// Figure out the status of the current channel.
+async fn channel_status(node: impl Borrow<Node>) -> Result<ChannelStatus> {
+    let node: &Node = node.borrow();
+    let node = &node.inner;
+
+    // We assume that the most recently created LN channel is the current one. We only care about
+    // that one because the app can only have one channel at a time.
+    let ln_channel = get_all_non_pending_channels()?
+        .iter()
+        .max_by(|a, b| a.created_at.cmp(&b.created_at))
+        .cloned();
+
+    let ln_channel = match ln_channel {
+        Some(ln_channel) => ln_channel,
+        // We never even had one LN channel.
+        None => return Ok(ChannelStatus::NotOpen),
+    };
+
+    let subchannels = node.list_dlc_channels()?;
+
+    let status = derive_ln_dlc_channel_status(ln_channel, subchannels);
+
+    Ok(status)
+}
+
+// TODO: We currently only look at the state of the subchannel. We should also take into account the
+// state of the DLC channel and the contract. Otherwise we won't be able to convey that the CET is
+// not yet confirmed.
+fn derive_ln_dlc_channel_status(
```

Comment:
🔧 this is a great function to cover with unit test to showcase how it behaves.

Reply:
I agree. I extracted it so that it could be tested, but I was too lazy at the time.

Reply:
I'm gonna be pragmatic here and say that we can add this unit test later because it's a pain to build the `SubChannel` coming from `rust-dlc`.

I think to make it more testable we need to map the foreign types to other module-private ones which only have the data we care about for `derive_ln_dlc_channel_status`.

Reply:
fair enough, if it's not trivial we can move it.
Maybe if we split this function into 2? keep the complex representation still internal to the module, but at least test the function that "simplifies" the state? 
(doesn't need to happen now).